documentation/modules/exploit/multi/http/apache_nifi_processor_rce.md

```diff
@@ -0,0 +1,207 @@
+    4. Replace `<property name="Manager Password"></property>` with `<property name="Manager Password">admin</property>`.
+    5. Replace `<property name="Url"></property>` with `<property name="Url">ldap://127.0.0.1</property>`.
+    5. Replace `<property name="User Search Base"></property>` with `<property name="User Search Base">dc=example,dc=org</property>`.
+    5. Replace `<property name="User Search Filter"></property>` with `<property name="User Search Filter">cn={0}</property>`.
+23. Restart NiFi by running  `./bin/nifi.sh restart`.
+24. NiFi will be available over HTTPS (**on the local machine only**) on port 9443 (be patient, NiFi takes some time to start).
+25. Ensure that you can access the web UI and authenticate using the username `admin` and password of `admin`. Remember that in step 12 you
+    may have needed to forward the port to the local machine. In order to be prompted for credentials, you must not authenticate using the
+    certificate like you did in step 12. You may have to take steps to prevent your browser from automatically submitting the certificate,
+    such as restarting the browser. A message will be displayed saying that you do not have permission to access the webUI. This is
+    expected, and indicates that NiFi does recognise the account.
+
+#### Useful links
+
+[NiFi Getting Started](https://nifi.apache.org/docs/nifi-docs/html/getting-started.html) (including installation on Windows/Linux/Unix).
+
+[NiFi Walkthroughs](https://nifi.apache.org/docs/nifi-docs/html/walkthroughs.html) (including installation on Unix and enabling HTTPS).
+
+[NiFi, Authentication and Authorization](https://ijokarumawak.github.io/nifi/2016/11/15/nifi-auth/) (configuring LDAP authentication).
+
+## Verification Steps
+1. Install the application.
+1. Start msfconsole.
+1. Do: `use exploit/multi/http/apache_nifi_processor_rce`.
+1. Do: `set rhosts <ip address of NiFi host> `.
+1. Do: `set lhost <ip address of metasploit machine`.
+1. If necessary, do: `set rport <port of NiFi API>`.
+1. If necessary, do: `set ssl true`.
+1. If necessary, do: `set username <NiFi username>`.
+1. If necessary, do: `set password <NiFi password>`.
+1. If necessary, do: `set target 1` (this sets the target to Windows instead of Unix).
+1. Do: `run`.
+1. You should get a shell.
+
+## Options
+
+### TARGETURI
+
+The base NiFi API path. Default: `/nifi-api`.
+
+### DELAY
+
+The delay in seconds before stopping and deleting the processor. This is required because the processor may not start immediately after
+being created. Default: `5`.
+
+### USERNAME
+
+Username to authenticate with, if necessary. Optional.
+
+### PASSWORD
+
+Password to authenticate with, if necessary. Optional.
+
+### BEARER-TOKEN
+
+JWT authenticate with, if necessary. Optional.
+
+## Scenarios
+The first target (Windows 10.0.18363 Pro) does not require authentication, however the second target (Ubuntu Server 20.04.1) does. It can be
```

Comment:
It might be worth splitting this into two separate scenario runs, one for each target

Reply:
Done in commit `1da912f`